ui/src/views/Tasks/TaskIndex/ListNamespaces/index.jsx

```diff
@@ -41,6 +43,30 @@ const defaultEmpty = defaultTo('');
   }),
 })
 export default class ListNamespaces extends Component {
+  state = {
+    namespaceInput: '',
+  };
+
+  componentDidMount() {
+    if (this.props.match.params.namespace) {
+      this.setState({ namespaceInput: this.props.match.params.namespace });
+    } else {
+      this.setState({ namespaceInput: '' });
+    }
+  }
```

Comment:
I wonder if this can be done in the state directly above to avoid an extra render.